### DISCUSSION TITLE
<lang>_args arguments for declare_dependency()

### DISCUSSION THREAD

**Discussion by georgev93** [mesonbuild/meson#11857] [Ideas]
Is there any reason why there couldn't be a <lang>_args argument for declare_dependency()? I use the declare_dependency() function for a lot of unit tested C libraries where the tests are written in C++, therefore, these unit tested modules often have both C and C++ compile args.

**Comment by eli-schwartz** [comment]
This is something that @dcbaker is trying to work on in the long term.

**Reply by dcbaker** [reply]
Yeah, this is something I'm working on. One thing I don't want to do is repeat the mistake we made with include directories (see: https://github.com/mesonbuild/meson/pull/10122) where `declare_dependency()` and `dependency()` do different things. So we need to fix all of the dependencies at once

**Comment by xclaesse** [comment]
I agree, `compile_args` is both confusing and broken, as far as I know that's just historical reasons that we should fix.
